bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -1008,3 +1022,17 @@ func (r *KubeadmConfigReconciler) storeBootstrapData(ctx context.Context, scope
 	conditions.MarkTrue(scope.Config, bootstrapv1.DataSecretAvailableCondition)
 	return nil
 }
+
+// reconcileCertificateExpiryTime reconciles the certificate expiry time on Kubeadmconfig.
+// Sets the certificate expiration time as on annotation on the config.
+// If the annotation is already present, there is nothing to do.
+func (r *KubeadmConfigReconciler) reconcileCertificateExpiryTime(_ context.Context, scope *Scope) {
```

Comment:
nit: can we pass only the config vs passing the entire scope; also, let's drop context if it not required